values/decoding/fs_time_values.go

```diff
@@ -36,19 +38,41 @@ func (v *fsBasedTimeValues) Metadata() values.TimeValuesMetadata {
 }
 
 func (v *fsBasedTimeValues) Iter() (iterator.ForwardTimeIterator, error) {
-	return newTimeIteratorFromMeta(v.metaProto, v.encodedValues)
+	return newTimeIteratorFromMeta(v.metaProto, v.encodedValues, true)
 }
 
-// TODO(xichen): Filter implementation should take advantage of the metadata
-// to do more intelligent filtering, e.g., checking if the value is within the
-// value range, and translate the filtering operation against the time values
+// TODO(xichen): Filter implementation should translate the filtering operation against the time values
 // into filtering operation against the underlying int values to take advantage
 // of a more optimized int value filter implementation.
 func (v *fsBasedTimeValues) Filter(
 	op filter.Op,
 	filterValue *field.ValueUnion,
 ) (iterator.PositionIterator, error) {
-	return defaultFilteredFsBasedTimeValueIterator(v, op, filterValue)
+	if filterValue == nil {
+		return nil, errNilFilterValue
+	}
+	if filterValue.Type != field.TimeType {
+		return nil, errUnexpectedFilterValueType
+	}
+	if !op.TimeIsInRange(v.metaProto, filterValue.TimeNanosVal) {
+		return impl.NewEmptyPositionIterator(), nil
+	}
+	// TODO(wjang): newTimeIteratorFromMeta below also calls protoResolutionToDuration. The overhead shouldn't
```

Comment:
`TODO:(wjang)` -> `NB(wjang)` unless you plan to fix this later.

Alternatively you can have a `newTimeIteratorWithResolution` that takes in a resolution and a meta.